torch/jit/__init__.py

```diff
@@ -631,12 +634,35 @@ def __getattr__(self, attr):
         return self.module._get_method(attr)
 
 
-def script(fn, optimize=True, _frames_up=0):
+def weak_script(fn, _frames_up=0):
+    compiled_weak_fns[fn] = {
+        "status": COMPILATION_PENDING,
+        "compiled_fn": None,
+        "rcb": createResolutionCallback(_frames_up + 1)
```

Comment:
discussed in person: modify createResolutionCallback to make sure it does not hold `frame` but instead holds only`f_locals` and `f_globals` to avoid keeping the entire stack of activations alive.